src/datapoint.jl

```diff
@@ -0,0 +1,61 @@
+## New Type: DataPoint
+# Struct for carrying statistic and systematic uncertainty of a DataPoint
+struct DataPoint{T<:Number} <: Number
+    val::T
+    stat_err::Tuple{T,T}
+    sys_err::Tuple{T,T}
+end
+
+# Helper functions
+_to_tuple(x::T where T<:Number) = (x,x)
+_to_tuple(x::Tuple) = promote(x...)
+_check_error_positivity(err) = all(x -> (π/2 >= angle(x)>=0), err)
+
+## Constructors
+# default constructor
+DataPoint() = DataPoint(NaN, (NaN, NaN), (NaN, NaN))
+# dispatch constructor
+DataPoint(x::Number, stat, sys = zero(x)) = DataPoint(x, stat = stat, sys = sys)
+# keyword constructor for integers (ceils uncertainties)
+# function DataPoint(x::Integer; stat=zero(x), sys=zero(x))
+#     @assert _check_error_positivity(stat) && _check_error_positivity(sys) "Only use positive uncertainties!"
+#     stat = _to_tuple(stat)
+#     sys = _to_tuple(sys)
+#     T = typeof(x)
+#     DataPoint(x,T.(ceil.(stat)),T.(ceil.(sys)))
+# end
```

Comment:
I think we can remove this code for now. We will likely never be in a situation where we want to uncomment this.

Reply:
You are probably right. 
Though for now it doesn't bother me if it stays inside?
Unless you have a very strong opinion here I would keep it at least until we finally merge this PR.

Reply:
Ok. Codecov is not complaining so let's keep it for now and get this merged.